### DISCUSSION TITLE
How to log non 2xx message?

### DISCUSSION THREAD

**Discussion by caibirdme** [apache/apisix#7835] [Q&A]
If got non 2xx response, I want to log it to stderr/stdout, is there any way to achieve this?

**Comment by tokers** [comment]
Access logs are always printed, it doesn't matter which the status code is.

**Reply by caibirdme** [reply]
I mean:
```
if response.http_status != 200 {
  log_detail(response)
}
```

**Reply by tzssangglass** [reply]
Still don't know what you're going to do.

Logging is independent of upstream status, and all upstream status is recorded in the log.

**Reply by caibirdme** [reply]
> Still don't know what you're going to do.
> 
> Logging is independent of upstream status, and all upstream status is recorded in the log.

For normal requests, we just need simple access log, something like `ts /a/b 200 ...`, we don't need to log the response body.
But if a request returns a non-2xx response, we may need not only the status code,  but also the response body to see why.
